iOS/Core/Telemetry/ProductSurfaceTelemetry.swift

```diff
@@ -0,0 +1,156 @@
+//
+//  ProductSurfaceTelemetry.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import BrowserServicesKit
+
+/// Product surface telemetry sends anonymous pixels about areas of the app that are being used so that we can we make future product decisions.  These pixels are not linked to any identifiable data.  The pixels are also enabled / disabled by config and we only enable them during the periods of product roadmap development in order to assist decision making.
+public protocol ProductSurfaceTelemetry {
+
+    ///  Called when the menu is used, either on new tab page or when browsing.
+    func menuUsed()
+
+    /// Daily active user indicator.
+    func dailyActiveUser()
+
+    /// Indicates the app is running on iPad.
+    func iPadUsed(isPad: Bool)
+
+    /// Indicates device is in landscape mode.
+    func landscapeModeUsed()
+
+    /// Indicates keyboard is focused (visible).
+    func keyboardActive()
+
+    /// Autocomplete surface used.
+    func autocompleteUsed()
+
+    /// Fires a pixel depending on the URL, either a search or a regular website
+    func navigationCompleted(url: URL?)
+
+    /// Duck.ai surface used.
+    func duckAIUsed()
+
+    /// Tab manager surface used.
+    func tabManagerUsed()
+
+    /// Data clearing from any source.
+    func dataClearingUsed()
+
+    /// New Tab Page surface used.
+    func newTabPageUsed()
+
+    /// Settings used (not re-fired on navigation within settings).
+    func settingsUsed()
+
+    /// Bookmarks page used.
+    func bookmarksPageUsed(isRootFolder: Bool)
+
+    /// Passwords page used.
+    func passwordsPageUsed()
+}
+
+public struct PixelProductSurfaceTelemetry: ProductSurfaceTelemetry {
```

Comment:
One suggestion here, to prevent having `guard featureFlagger.isFeatureOn(.productTelemeterySurfaceUsage) else { return }` in each function, is to create a no-op instance during launching when the flag is off. No need to pass feature flagger as a dependency too, and it would prevent sending only a part of events when flag was changed during runtime.